nodedb.go

```diff
@@ -763,13 +767,16 @@ func (ndb *nodeDB) GetRoot(version int64) ([]byte, error) {
 		if val == nil {
 			return nil, ErrVersionDoesNotExist
 		}
+		if len(val) == 0 { // empty root
+			return nil, nil
+		}
 		return val, nil
 	}
 	if len(val) == 0 { // empty root
 		return nil, nil
 	}
 	if isReferenceToRoot(val) { // point to the prev root
-		return val[1:], nil
+		return append(val[1:], 0, 0, 0, 1), nil
```

Comment:
what is this change for?

Reply:
previously, we were storing the node key as a reference node, now we are only storing the reference version without the nonce, since the root nonce is always 1